File: advent-js/2024/7-the-grinchs-attack/__main__.py
def fix_packages(packages):
    queue = [""]
    for letter in packages:
        if letter == "(":
            queue.append("")
        elif letter == ")":
            last = queue.pop()[::-1]
            queue[-1] += last
        else:
            queue[-1] += letter
    return queue[0]

File: advent-js/2024/7-the-grinchs-attack/main.js
/** @param {string} packages with parentheses
 *  @returns {string} Fixed and sorted packages
 */
function fixPackages(packages) {
  let queue = ['']
  for (let letter of packages) {
    if (letter === '(') {
      queue.push('')
      continue
    }
    if (letter === ')') {
      const last = queue.pop().split('').reverse().join('')
      queue[queue.length - 1] += last
      continue
    }
    queue[queue.length - 1] += letter
  }
  return queue[0]
}
